### ISSUE TITLE
Add content-length http header to response

### ISSUE THREAD

**Issue by johnlund** [wp-graphql/wp-graphql#1285]
Hi WPGraphQL,

I'm trying to create a percentage-based loading bar on my front-end, and I'm using axios to get access to the request progressEvent with the hopes of getting a `total` value that I can use to base my loading percentage on. But the response from wp-graphql does not have a content-length header, so the `total` value is 0.

I'm not sure how complicated adding this header is though. I'm happy to work on this if this is something you feel would be beneficial. I would just need a little direction. :)

Thanks!
John

**Comment by jasonbahl** [comment]
@johnlund I'm not fully opposed to this idea. Looking around at many other GraphQL servers, I don't see this header being returned though, so I'm not sure how common it is to implement. 

My thought is that if you're getting slow enough responses that you need a loader, you might want to consider limiting your requests to what you actually need. 

Do you have an example of a Query that is taking so long you feel like a loading bar is necessary?

Are you feeling like the download time is taking some time? You make a request and it has so much info that downloading is a bit of a bottleneck?

**Comment by johnlund** [comment]
@jasonbahl Sorry for the delayed response.

The WPGraphQL query was not taking a particularly long time to load. I was trying to build a progress bar to take the loading of the entire page. But after I wrote that request, I realized that many resources are not returning this header, so the import of it being included in the WPGraphQL response has dwindled for me. Without all these headers, I just decided to use a timed animation instead of a calculated bar.

So, I think I agree with you in that it doesn't seem as common of an implementation as I originally thought, and though I could see some value in it, I don't know if it's worth the effort.
